presto-main/src/main/java/com/facebook/presto/testing/MaterializedRow.java

```diff
@@ -53,8 +53,11 @@ public MaterializedRow(int precision, List<Object> values)
     private static Object processValue(int precision, Object value)
     {
         checkArgument(!(value instanceof Short || value instanceof Byte), "SMALLINT and TINYINT are not yet supported");
-        if (value instanceof Double || value instanceof Float) {
-            return new ApproximateDouble(((Number) value).doubleValue(), precision);
+        if (value instanceof Double) {
```

Comment:
Why have ApproximateNumeric at all? Why not just have ApproximateFloat and ApproximateDouble?